### ISSUE TITLE
Hot reloading config does not update password

### ISSUE THREAD

**Issue by Will2817** [nats-io/gnatsd#926]
- [x] Defect
 - [ ] Feature Request or Change Proposal
 
#### Versions of `gnatsd` and affected client libraries used:
Tested for defect with v1.3.0 and v1.4.1

#### OS/Container environment:
MacOS, Docker, Kubernetes

#### Steps or code to reproduce the issue:
Starts gnatsd with `-c` flag to the following config file
```
authorization {
      user: nats_client
      password: password1
}
```
Attempt to create client connection to verify user/password is configured:
```
telnet localhost 4222
CONNECT{"user":"nats_client","pass":"password1"}
```
You should get an `OK` response from the server

Update config to:
```
authorization {
      user: nats_client
      password: password2
}
```
and run the following command to reload the config file:
```
gnatsd -sl reload=<running process-id>
```

#### Expected result:

Attempting to authenticate with the above method with `password1` should fail, and `password2` should succeed.
 
#### Actual result:
 
Using `password1` for authentication succeeds and "password2" fails.

**Comment by derekcollison** [comment]
Do you have multiple servers running? Will double check but we have quite a few tests around this as you would imagine.

**Comment by Will2817** [comment]
Hi @derekcollison just one server. And absolutely, it's totally possible I just misconfigured something, or I'm misunderstanding how config reloading is suppose to work, but I couldn't seem to get this to work.

**Comment by derekcollison** [comment]
ok I just tested it locally on my mac and works as it is supposed to. "password1" no longer works and "password2" does work.

So something else is going on..

Where is the server running?
Where are you running the telnet session?

**Comment by kozlovic** [comment]
What is strange is that if this is your exact config, you would be unlikely to even test it with telnet since your telnet session should get disconnected with a auth timeout (default is 0.5 sec). So maybe you are not telnetting to the correct server?

**Comment by derekcollison** [comment]
Have the CONNECT{} in the copy buffer. Up arrow to telnet command and hit enter and paste <enter>. Works fine. You can increase timeout if you want. You can also run the server with -DV on the command line so it prints out more information etc.

Trust me this functionality works as it is supposed to.

**Comment by Will2817** [comment]
I'm running both the server and telnet on locally on my mac. @kozlovic I am able to send the CONNECT command fast enough to beat 0.5 sec timeout.

**Comment by kozlovic** [comment]
Works for me with downloaded 1.3.0 and 1.4.1 releases. Can you run server 1.4.1 with `-DV` and provide output including the config reload outcome? Note that in 1.3.0 had a bug where if you run with `-DV` after the reload, the debug and trace would be reset to false. If you run 1.3.0, use `-D -V` instead.

**Comment by Will2817** [comment]
I found my mistake, I had originally found the issue with a version of gnatsd with CustomClientAuthentication, and then went to confirm the issue with an official gnatsd binary, and I must have done something wrong when switching binary, because the official binary seems to work fine for me now. 

For clarity sake, my mistake with custom auth is that it has it's own list of clients credentials (including those from the config file) and is not being updated when a reload of the config happens. Which will be whole other problem to try and solve 😄.

Thanks @derekcollison and @kozlovic for your help.

Closing based on user error 😅.
